### ISSUE TITLE
BUG: Gas required exceeds allowance (550000000)

### ISSUE THREAD

**Issue by AdamK222** [ethereum-optimism/optimism#5678]
**Describe the bug**
"Gas required exceeds allowance" error coming from RPC whenever I attempt to deploy a modified ERC-20 token on Optimism using remix IDE. Pure ERC-20 token works flawlessly, but the one which calls additional contracts in its constructor is showing that error and prevents deployment.

**To Reproduce**
1. https://pastebin.com/ecGjemi2 - use this code for ERC-20 token, the most important part is _updateRouterV2(0x744776F27080b584D447A780ba260c435f3aE7d5); which I think causes the error
3. Paste it in remix and compile with Optimizer enabled, 200 runs
4. Preferably use Injected MetaMask provider to attempt the deployment
5. See this error 
![image](https://github.com/ethereum-optimism/optimism/assets/66825147/648f6cf7-0ced-4967-9b66-1d3313b64278)
6. After force-sending the transaction fails by spending full 3M of gas, however successful deployment will require 4.3M of gas (tried the same on BSC)

**Expected behavior**
Contract should deploy without errors with correctly adjusted gas limit

**Comment by smartcontracts** [comment]
Usually this happens when there's a revert in the constructor that always gets thrown. Likely not an issue with OP Mainnet.

**Comment by AdamK222** [comment]
I tested this on 7 different blockchains with around 20 different DEXes' addresses included in constructor and not only on all other blockchains is succeeds but also no matter which DEX I put on optimism it always fail and always with the same error. Because I'm using Metamask I'm not able to increase the gas limit for deployment tx.

Is Optimism running with hardcoded 3M gas limit hard cap ?

**Comment by 0xManaria** [comment]
same problem

**Comment by AdamK222** [comment]
I experimented a bit more and now I can give a much clearer example, maybe this will help more.

Deployment of this will fail:
![image](https://github.com/ethereum-optimism/optimism/assets/66825147/7ee80ce9-0486-43f8-85ef-64ad70b69b52)

But if I change the `address(this)` to token's address which already exists on blockchain deployment will succeed:
![image](https://github.com/ethereum-optimism/optimism/assets/66825147/0ff58ae3-a75a-416b-a672-d9acb3cd8af1)

I also did some debugging and found that it fails in UniswapV2Pair contract being created as a result of createPair() function call. It fails precisely on 76 or 77 line when it is making a call to fetch the symbol of token being deployed:
![image](https://github.com/ethereum-optimism/optimism/assets/66825147/ad24818b-fb1f-4a64-97eb-f5413500f8ef)

I searched the "Differences between Ethereum and Optimism" article but I didn't find out why is this happening. What is the difference between this 2 blockchains that makes external call to revert if token is being deployed in the same transaction?
